src/components/v5/common/CompletedAction/partials/SimplePayment/SimplePayment.tsx

```diff
@@ -57,14 +60,15 @@ const SimplePayment = ({ action }: SimplePaymentProps) => {
         {formatText(MSG.subtitle, {
           amount: formattedAmount,
           token: action.token?.symbol,
-          recipient: recipientUser ? (
+          recipient: recipientAddress ? (
             <UserPopover
-              userName={recipientUser.profile?.displayName}
-              walletAddress={recipientUser.walletAddress}
+              userName={recipientUser?.profile?.displayName}
```

Comment:
do we need to change the `userName` to be the address if the profile doesn't exist?

Reply:
Correct! If the user hasn't registered a profile yet, then we should show the address, masked or not. This applies to every other place in the CDapp.